### ISSUE TITLE
Azure not accepting the most recent changes

### ISSUE THREAD

**Issue by itsn0tm3** [nightscout/cgm-remote-monitor#2929]
Getting this error when trying to build on Azure:
(Have already put the Node string to point to 8.5.0, and tried building both Dev and Master, and both do not work)
```
Command: bash deploy.sh
Handling node.js deployment.
KuduSync.NET from: 'D:\home\site\repository' to: 'D:\home\site\wwwroot'
Copying file: 'scm-commit-id.json'
Using start-up script server.js from package.json.
Node.js versions available on the platform are: 0.6.20, 0.8.2, 0.8.19, 0.8.26, 0.8.27, 0.8.28, 0.10.5, 0.10.18, 0.10.21, 0.10.24, 0.10.26, 0.10.28, 0.10.29, 0.10.31, 0.10.32, 0.10.40, 0.12.0, 0.12.2, 0.12.3, 0.12.6, 4.0.0, 4.1.0, 4.1.2, 4.2.1, 4.2.2, 4.2.3, 4.2.4, 4.3.0, 4.3.2, 4.4.0, 4.4.1, 4.4.6, 4.4.7, 4.5.0, 4.6.0, 4.6.1, 4.8.4, 5.0.0, 5.1.1, 5.3.0, 5.4.0, 5.5.0, 5.6.0, 5.7.0, 5.7.1, 5.8.0, 5.9.1, 6.0.0, 6.1.0, 6.2.2, 6.3.0, 6.5.0, 6.6.0, 6.7.0, 6.9.0, 6.9.1, 6.9.2, 6.9.4, 6.9.5, 6.10.0, 6.10.3, 6.11.1, 6.11.2, 7.0.0, 7.1.0, 7.2.0, 7.3.0, 7.4.0, 7.5.0, 7.6.0, 7.7.0, 7.7.4, 7.10.0, 7.10.1, 8.0.0, 8.1.4, 8.4.0, 8.5.0, 8.7.0.
Selected node.js version 8.5.0. Use package.json file to choose a different version.
Selected npm version 5.3.0
Updating iisnode.yml at D:\home\site\wwwroot\iisnode.yml
npm WARN invalid config scripts-prepend-node-path="--production"
npm ERR! path D:\home\site\wwwroot\node_modules\fsevents\node_modules\getpass\node_modules
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall scandir
An error has occurred during web site deployment.
npm ERR! Error: EPERM: operation not permitted, scandir 'D:\home\site\wwwroot\node_modules\fsevents\node_modules\getpass\node_modules'
npm failed
npm ERR!  { Error: EPERM: operation not permitted, scandir 'D:\home\site\wwwroot\node_modules\fsevents\node_modules\getpass\node_modules'
npm ERR!   stack: 'Error: EPERM: operation not permitted, scandir \'D:\\home\\site\\wwwroot\\node_modules\\fsevents\\node_modules\\getpass\\node_modules\'',
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'scandir',
npm ERR!   path: 'D:\\home\\site\\wwwroot\\node_modules\\fsevents\\node_modules\\getpass\\node_modules' }
npm ERR! 
npm ERR! Please try running this command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     D:\local\AppData\npm-cache\_logs\2017-10-17T05_33_21_572Z-debug.log
npm WARN invalid config scripts-prepend-node-path="--production"\r\nnpm ERR! path D:\home\site\wwwroot\node_modules\fsevents\node_modules\getpass\node_modules\r\nnpm ERR! code EPERM\r\nnpm ERR! errno -4048\r\nnpm ERR! syscall scandir\r\nnpm ERR! Error: EPERM: operation not permitted, scandir 'D:\home\site\wwwroot\node_modules\fsevents\node_modules\getpass\node_modules'\r\nnpm ERR!  { Error: EPERM: operation not permitted, scandir 'D:\home\site\wwwroot\node_modules\fsevents\node_modules\getpass\node_modules'\r\nnpm ERR!   stack: 'Error: EPERM: operation not permitted, scandir \'D:\\home\\site\\wwwroot\\node_modules\\fsevents\\node_modules\\getpass\\node_modules\'',\r\nnpm ERR!   errno: -4048,\r\nnpm ERR!   code: 'EPERM',\r\nnpm ERR!   syscall: 'scandir',\r\nnpm ERR!   path: 'D:\\home\\site\\wwwroot\\node_modules\\fsevents\\node_modules\\getpass\\node_modules' }\r\nnpm ERR! \r\nnpm ERR! Please try running this command again as root/Administrator.\r\n\r\nnpm ERR! A complete log of this run can be found in:\r\nnpm ERR!     D:\local\AppData\npm-cache\_logs\2017-10-17T05_33_21_572Z-debug.log\r\nD:\Program Files (x86)\SiteExtensions\Kudu\66.61008.3066\bin\Scripts\starter.cmd bash deploy.sh
```

**Comment by itsn0tm3** [comment]
Also when trying to redeploy I get this "Failed to redeploy deployment. Heuristics indicate WebApiClient request timed out. "

**Comment by itsn0tm3** [comment]
Didn't realize there was a URI with the above message and it says `{"error":{"code":"AuthenticationFailed","message":"Authentication failed. The 'Authorization' header is missing."}}`

**Comment by barryhelsden** [comment]
I'm having the same issue. Happy to provide logs if it helps.

**Comment by sulkaharo** [comment]
This sounds like you’ve found an Azure bug that even some Microsoft employees are bumping into. Ping @shanselman. The only known fix is to delete the github connection from your Azure account and reconnect & deploy. Alternatively I recommend looking at Heroku for the hosting, which is much nicer to use and has none of these bugs.

**Comment by barryhelsden** [comment]
Update - working for me now, I followed the troubleshooting part 2 steps here: 

http://www.nightscout.info/wiki/welcome/how-to-update-to-latest-cgm-remote-monitor-aka-cookie/update-my-fork-troubleshooting-part-2

Reset WEBSITE_NODE_DEFAULT_VERSION to 8.5.0, redeployed and its now working for me.

**Comment by MilosKozak** [comment]
This helps if automatic deploy fail:
go to https://YOURSITE.scm.azurewebsites.net/DebugConsole
```
cd site
cd wwwroot
del /f /s /q node_modules
rmdir /s /q node_modules
npm install
node server.js
```

**Comment by itsn0tm3** [comment]
If I do those steps, will it make my current deployment stop running?  Or can I safely do those and then try to redeploy the newer version?

**Comment by itsn0tm3** [comment]
Moved over to Heroku to solve this.

Closing.

**Comment by itsn0tm3** [comment]
So, Heroku went down so I had to move back to Azure while Heroku was down.  Looked into fixing the deployment and made one change that seems to have made the difference.  To the very right of the setting I have a checkbox option called "Slot Setting" which was previously unchecked for all of my settings and I've never had to use it.  I went ahead and checked the box and then saved it.  Went ahead and tried the redeploy and now it is building successfully.

Not quite sure why that made a difference though.
